pdm/installers/installers.py

```diff
@@ -25,6 +25,85 @@
 
     from installer.destinations import Scheme
 
+    from pdm.models.environment import Environment
+
+
+@lru_cache()
+def _is_python_package(root: str) -> bool:
+    for child in Path(root).iterdir():
+        if child.is_file():
+            if child.suffix in (".py", ".pyc", ".pyo", ".pyd"):
+                return True
+        else:
+            if child.joinpath("__init__.py").exists():
+                return True
+    return False
+
+
+@lru_cache()
+def _is_namespace_package(root: str) -> bool:
+    if not _is_python_package(root):
+        return False
+    if not os.path.exists(os.path.join(root, "__init__.py")):  # PEP 420 style
+        return True
+    int_py_lines = [
+        line.strip()
+        for line in Path(root, "__init__.py").open(encoding="utf-8")
+        if line.strip() and not line.strip().startswith("#")
+    ]
+    namespace_identifiers = [
+        # pkg_resources style
+        "__import__('pkg_resources').declare_namespace(__name__)",
+        # pkgutil style
+        "__path__ = __import__('pkgutil').extend_path(__path__, __name__)",
+    ]
+    checker = namespace_identifiers[:]
+    checker.extend(item.replace("'", '"') for item in namespace_identifiers)
+    return any(line in checker for line in int_py_lines)
+
+
+def _create_symlinks_recursively(source: str, destination: str) -> Iterable[str]:
+    """Create symlinks recursively from source to destination.
+    Caveats: This don't work for pkgutil or pkg_resources namespace packages.
+    package  <-- link
+        __init__.py
+    namespace_package  <-- mkdir
+        foo.py  <-- link
+        bar.py  <-- link
+    """
+    for root, dirs, files in os.walk(source):
+        is_top = os.path.normcase(os.path.realpath(source)) == os.path.normcase(
+            os.path.realpath(root)
+        )
+        bn = os.path.basename(root)
+        if bn == "__pycache__" or bn.endswith(".dist-info"):
+            dirs[:] = []
+            continue
+        relpath = os.path.relpath(root, source)
+        destination_root = os.path.join(destination, relpath)
+        if not is_top and not _is_namespace_package(root):
+            # A package, create link for the parent dir and don't proceed
+            # for child directories
+            if os.path.exists(destination_root):
+                os.remove(destination_root)
+            os.symlink(root, destination_root, True)
+            yield relpath
+            dirs[:] = []
+            continue
+        # Otherwise, the directory is likely a namespace package,
+        # mkdir and create links for all files inside.
```

Comment:
Very nice1 From glancing at the code, it seems you support nested namespace packages?

Reply:
Yeah, but I guess it won't work if the namespace packages are nested in more than 2 levels:

```
a   # <- this will be detected as normal directory and create link for it
    b
        c
           __init__.py
```

But I think it is enough.

Reply:
Ah I see. Too bad, I actually have a doubly nested namespace package 😅 
But it's OK, pth files work fine, and I'm planning to change that to a level-1 namespace package later anyway.
Thanks!

Reply:
it is easy for a recursive detection, but i am not sure if a deeply nested package can still be called a 'package'. However, if you agree every py files inside a wheel must belong to a package, the answer would be Yes.

Reply:
Well, that's how PEP420 packages work, so I'd say yes, with `a/b/c/d/e.py`, `a` is a (namespace) package (and b, c, and d are sub-namespace packages, or namespace sub-packages haha).